packages/lib/gpc/src/gpcTypes.ts

```diff
@@ -63,6 +69,8 @@ export type GPCProofEntryConfig = {
    */
   equalsEntry?: PODEntryIdentifier;
 
+  liesInLists?: PODName[];
```

Comment:
The reason I stuck this in the entries and tuples is precisely to allow checking for membership in multiple lists and to allow multiple entry values to be members of a given list. Are these cases we didn't want to allow?

Reply:
You actually coded this to be more flexible than I expected when I reviewed the interface.  I don't think there's any reason to disallow anything the code can reasonably support.  It seems a little bit obfuscated to be taking 1 input list and duplicating it into 2 lists in the circuit inputs.  What it's hiding is an impact on circuit+signal size, though now that I think about it the alternative of forcing the user to manually duplicate their list is probably worse, since it affects PCD size, and also hides the fact that the two lists are equal.

I'm convinced.  I do think this deserves a good paragraph somewhere to explain how the inputs map to the limits in a CircuitDescription.  E.g. the fact that you can have an input with NxM lists, which doesn't fit into a circuit with NxM lists (because you're checking 2 entries per list) will be unintuitive.  Somewhere in the docs need to explain that the number in the circuit is a number of list membership checks (1 input to 1 list) without any reusability.

(As an aside, we could build a circuit to reuse one list of inputs for multiple membership checks at the cost of some more constraints, though I'm not sure if it's worth it for what's likely to be a rare case.  nConstraints seems like a stronger optimization target than nInputs.)